pkg/scheduler/framework/plugins/schedulinggates/scheduling_gates.go

```diff
@@ -59,26 +57,17 @@ func (pl *SchedulingGates) PreEnqueue(ctx context.Context, p *v1.Pod) *framework
 // EventsToRegister returns the possible events that may make a Pod
 // failed by this plugin schedulable.
 func (pl *SchedulingGates) EventsToRegister() []framework.ClusterEventWithHint {
-	if !pl.enableSchedulingQueueHint {
```

Comment:
Wouldn't this cause code that should be feature gated to always run?

Reply:
This is aligned with other plugins, always register the events whether `enableSchedulingQueueHint` is enabled or not. 

We should avoid to register empty events now with https://github.com/kubernetes/kubernetes/pull/125527 merged, or it will lead to pod stuck when `enableSchedulingQueueHint` enabled, both `movePodsToScheduilngQueue` and `podUpdate` are depend on the queueingHint right now.

Reply:
If the feature gate is disabled, empty event in `EventsToRegister` doesn't cause Pod stuck because the change introduced by https://github.com/kubernetes/kubernetes/pull/122234 is feature-gated, does it?

Reply:
If the feature gate is enabled, which is likely future, it does, like you said. 
So, I agree that we should encourage people avoid empty event, though.

Reply:
You're right, the only impact lies at: With PodUpdateEvent in schedulingQueue, feature gate enabled but we registered empty events, which may lead to Pod stuck because we're always returning queueSkip status.

Reply:
> feature gate enabled but we registered empty events

why would it register empty events, if the feature gate is enabled?

Reply:
We didn't restrict this behavir.

Reply:
I believe we have two options:

1. Keep the current implementation: The current implementation has to return Pod/Update event from the scheduilng queue because otherwise the warning log we added in pkg/scheduler/scheduler.go emits always for default scheduler.
2. Revert the change in scheduling_gates.go and feature-gate the warning log.

I'm fine with (1), but leave `/approve` to Aldo.

/lgtm

Reply:
ping @alculquicondor

Reply:
just bear with me a little longer. I have too many reviews in kueue and scheduler.

Reply:
I feel you.

Reply:
I prefer 2, so that we restrict this change under the feature gate altogether.

Reply:
Forgot this one, I'll follow the stricter way, forbidden to register empty events. I think still makes sense here.